examples/vision/depth_estimation.py

```diff
@@ -0,0 +1,489 @@
+"""
+
+annotation_folder = "/dataset/"
+if not os.path.exists(os.path.abspath(".") + annotation_folder):
+    annotation_zip = tf.keras.utils.get_file(
+        "val.tar.gz",
+        cache_subdir=os.path.abspath("."),
+        origin="http://diode-dataset.s3.amazonaws.com/val.tar.gz",
+        extract=True,
+    )
+
+"""
+##  Preparing the dataset
+"""
+
+path = "val"
+
+filelist = []
+
+for root, dirs, files in os.walk(path):
+    for file in files:
+        filelist.append(os.path.join(root, file))
+
+filelist.sort()
+data = {
+    "image": [x for x in filelist if x.endswith(".png")],
+    "depth": [x for x in filelist if x.endswith("_depth.npy")],
+    "mask": [x for x in filelist if x.endswith("_depth_mask.npy")],
+}
+df = pd.DataFrame(data)
+
+df = df.sample(frac=1, random_state=42)
+
+"""
+## Prepare Hyperparameters
+"""
+
+HEIGHT = 256
+WIDTH = 256
+LR = 0.0002
+EPOCHS = 30
+BATCH_SIZE = 32
+
+"""
+## Building Dataset Loader Pipeline
+
+1. The pipeline takes dataframe mainintaing the path for RGB, depth and depth mask files
+as input.
+2. Reads and resize the RGB images
+3. Reads the depth and depth_mask files, process them to generate the depth-map image and
+resize it.
+4. Returns the RGB images and the depth-map images for a batch.
+"""
+
+
+class DataGenerator(tf.keras.utils.Sequence):
+    def __init__(self, data, batch_size=6, dim=(768, 1024), n_channels=3, shuffle=True):
+        """
+        Initialization
+        """
+        self.data = data
+        self.indices = self.data.index.tolist()
+        self.dim = dim
+        self.n_channels = n_channels
+        self.batch_size = batch_size
+        self.shuffle = shuffle
+        self.on_epoch_end()
+
+    def __len__(self):
+        return int(np.ceil(len(self.data) / self.batch_size))
+
+    def __getitem__(self, index):
+        if (index + 1) * self.batch_size > len(self.indices):
+            self.batch_size = len(self.indices) - index * self.batch_size
+        # Generate one batch of data
+        # Generate indices of the batch
+        index = self.indices[index * self.batch_size : (index + 1) * self.batch_size]
+        # Find list of IDs
+        batch = [self.indices[k] for k in index]
+        X, y = self.__data_generation(batch)
+
+        return X, y
+
+    def on_epoch_end(self):
+
+        """
+        Updates indexes after each epoch
+        """
+        self.index = np.arange(len(self.indices))
+        if self.shuffle == True:
+            np.random.shuffle(self.index)
+
+    def __load__(self, image_path, depth_map, mask):
+        "Load Input and Target image"
+
+        image_ = cv2.imread(image_path)
+        image_ = cv2.cvtColor(image_, cv2.COLOR_BGR2RGB)
+        image_ = cv2.resize(image_, self.dim)
+        image_ = tf.image.convert_image_dtype(image_, tf.float32)
+
+        depth_map = np.load(depth_map).squeeze()
+
+        mask = np.load(mask)
+        mask = mask > 0
+
+        MIN_DEPTH = 0.5
```

Comment:
This seems like it should be a (lowercase) instance attribute defined in `__init__()`